src/script/utils/icons.ts

```diff
@@ -0,0 +1,60 @@
+import { Icon } from "./interfaces";
+
+/**
+   * Finds an icon matching the specified purpose and desired dimensions.
+   * @param mimeType Should be an image mime type, e.g. "image/png", or null or empty to ignore format.
+   */
+  export function findSuitableIcon(
```

Comment:
Nice! I have some ideas for some logic reuse, I'll add some stuff here, but sorta like we're doing with the manifest, it might be nice to have a reference to the edited list of icons for easy querying, editing, and etc.